File: Hunt_Career_TestNG_Hybrid_Framework/src/main/java/com/huntcareer/qa/pages/EmployerRegisterPage.java
package com.huntcareer.qa.pages;

import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.support.FindBy;
import org.openqa.selenium.support.PageFactory;

public class EmployerRegisterPage extends PageBase {

    @FindBy(xpath = "//input[@id='companyName']")
    private WebElement companyNameFeild;

    @FindBy(xpath = "//input[@id='email']")
    private WebElement emailField;

    @FindBy(xpath = "//input[@id='password']")
    private WebElement passwordField;

    @FindBy(xpath = "//input[@id='confirmPassword']")
    private WebElement confirmPasswordField;

    @FindBy(xpath = "//button[@type='submit']")
    private WebElement signUpButton;

    @FindBy(xpath = "//div[contains(text(),'Employer already exists')]")
    public WebElement duplicateErrorMessage;

    @FindBy(xpath = "//div[contains(text(),'Company Name is required.')]")
    public WebElement companyNameBlankMessage;

    @FindBy(xpath = "//div[contains(text(),'Email is required.')]")
    public WebElement emailBlankMessage;

    @FindBy(xpath = "//div[contains(text(),'Please enter a valid email address.')]")
    public WebElement pleaseEnterValidEmailAddressMessage;

    @FindBy(xpath = "//div[contains(text(),'Password must be at least 8 characters long.')]")
    public WebElement passwordMustHaveRequiredNumberCharactersMessage;

    @FindBy(xpath = "//div[contains(text(),'Passwords do not match.')]")
    public WebElement passwordNotMatchMessage;

    @FindBy(xpath = "//a[contains(text(), 'Sign up')]")
    private WebElement SignupLink;

    public EmployerRegisterPage(WebDriver driver) {
        super(driver);
        PageFactory.initElements(driver, this);
    }

    public void enterCompanyName(String companyNameText) {
        typeInElement(companyNameFeild, companyNameText);
    }

    public void enterEmail(String emailText) {
        typeInElement(emailField, emailText);
    }

    public void enterPassword(String passwordText) {
        typeInElement(passwordField, passwordText);
    }

    public void enterConfirmPassword(String confirmPasswordText) {
        typeInElement(confirmPasswordField, confirmPasswordText);
    }

    public void clickSignupLink() {
        clickElement(SignupLink);
    }

    public EmployerLoginPage clickSignUpButton() {
        clickElement(signUpButton);
        return new EmployerLoginPage(driver);
    }
}

File: Hunt_Career_Playwright/pages/EmployerRegisterPage.js
import { BasePage } from "./BasePage";

export class EmployerRegisterPage extends BasePage {
  constructor(page) {
    super(page);
    this.getNameField = page.locator("//input[@id='companyName']");
    this.getEmailField = page.locator("#email");
    this.getPasswordField = page.locator("#password");
    this.getConfirmPasswordField = page.locator("#confirmPassword");
    this.getSignupButton = page.locator("//button[@type='submit']");
    this.getDuplicateErrorMessage = page.locator(
      "//div[contains(text(),'Employer already exists')]",
    );
    this.getNameBlankMessage = page.locator(
      "//div[contains(text(),'Company Name is required.')]",
    );
    this.getInvalidEmailAddressMessage = page.locator(
      "//div[contains(text(), 'Please enter a valid email address.')]",
    );
    this.getEmailBlankMessage = page.locator(
      "//div[contains(text(), 'Email is required.')]",
    );
    this.getPasswordLengthErrorMessage = page.locator(
      "//div[contains(text(), 'Password must be at least 8 characters long.')]",
    );
    this.getPasswordMismatchMessage = page.locator(
      "//div[contains(text(), 'Passwords do not match.')]",
    );
    this.getRegisterLinkHomePage = page.locator("//a[contains(text(), 'Sign up')]");
  }

  async navigate() {
    await this.navigateToEmployer();
    await this.clickElement(this.getRegisterLinkHomePage);
  }

  async register(
    name,
    email,
    password,
    confirmPassword,
  ) {
    await this.typeInElement(this.getNameField, name);
    await this.typeInElement(this.getEmailField, email);
    await this.typeInElement(this.getPasswordField, password);
    await this.typeInElement(this.getConfirmPasswordField, confirmPassword);
    await this.clickElement(this.getSignupButton);
  }
}
